client/components/social-buttons/qr-code.tsx

```diff
@@ -17,17 +16,12 @@ type QrCodeLoginButtonProps = {
 	loginUrl: string;
 };
 
-export const QrCodeLoginButton = ( { loginUrl }: QrCodeLoginButtonProps ) => {
+export default function QrCodeLoginButton( { loginUrl }: QrCodeLoginButtonProps ) {
 	const dispatch = useDispatch();
 	const translate = useTranslate();
-	const { isDisabled, oauth2Client, isWoo } = useSelector( ( select ) => {
-		return {
-			oauth2Client: getCurrentOAuth2Client( select ) as { id: string },
-			locale: getCurrentLocaleSlug( select ),
-			isWoo: getIsWoo( select ),
-			isDisabled: isFormDisabled( select ),
-		};
-	} );
+	const isDisabled = useSelector( isFormDisabled );
```

Comment:
Thank you for this! I think we need to create an ESLint rule to prevent this kind of misuse.

Reply:
We have a runtime warning that catches this problem. Unfortunately we're very far from a zero-warning dev console.

And there is also a simple way how to catch it with static code analysis: if an inline selector returns object or array literal, it's bad. Doesn't catch all bugs, but on the other hand doesn't have any false positives, it's reliable.